code/modules/keybindings/_readme.dm

```diff
@@ -0,0 +1,38 @@
+/*
+	This whole system is heavily based off of forum_account's keyboard library. As in I copied and
+	pasted it and then tweaked it to fit our needs. So thanks to forum_account for saving the day.
+
+	.dmf macros have some very serious shortcomings. For example, they do not allow reusing parts
+	of one macro in another, so giving cyborgs their own shortcuts to swap active module couldn't
+	inherit the movement that all mobs should have anyways. The webclient only supports one macro,
+	so having more than one was problematic. Additionally each keybind has to call an actual
+	verb, which meant a lot of hidden verbs that just call one other proc. Also our existing
+	macro was really bad and tied unrelated behavior into Northeast(), Southeast(), Northwest(),
+	and Southwest().
+
+	The basic premise of this system is to not screw with .dmf macro setup at all and handle
+	pressing those keys in the code instead. We have every key call client.keyDown()
+	or client.keyUp() with the pressed key as an argument. Certain keys get processed
+	directly by the client because they should be doable at any time, then we call
+	keyDown() or keyUp() on the client's holder and the client's mob's focus.
```

Comment:
the issue is the muddy hacky way you'd have to do it, in order to set client side macros, you'd have to hook together a hacky string containing all macros for client side use and send them to the client, all changes would requiring re-initializing the client's skin's macro string, meaning you can't send an add command, you have to track all client side macros and send them.

By doing this, you would also blind server side code to that macro, so now you can't make certain mobs override the that key's usage (the whole point of this system).

and just overall it would require tracking 3 separate settings, one for what the client currently has, one for what the client is suppose to have, and the normal key binding system.

it COULD be done with winset/winget logic, but its just hacky as fuck and i don't blame mrP for not wanting to walk down that line.